### PR TITLE
fix: navigation bar color management

### PR DESCRIPTION
**Pull request by kirillzyusko**
### Explanation of Change

Previously we didn't specify `navigationBarColor` at all. It was working well on some devices such as Pixel that manages `navigationBarColor` in a consistent way (on Pixel devices this `navigationBar` will be always dark, no matter which device theme is set).

However some device manufacturers are managing it differently, for example Samsung - they make `navigationBar` to match the color of the device theme (i. e. it will be white if device theme is "light", and will be black if device theme is "dark"). While it sounds pretty logical it produces conflicts in Expensify app: if device theme is "light" but user selected "dark" theme in Expensify app, then the app will be shown in dark mode, but navigation bar will be white. This is exactly what we got in: https://github.com/Expensify/App/issues/51673

From above it's clear, that we should manage `navigationBarColor` inside the app and don't rely on default OS management as it may vary depending on manufacturers.

This is what this PR brings. Mainly it affects 3 areas (to match closely to iOS):
- when splash screen is shown, then `navigationBar` color should match splash screen color (i. e. to be green);
- login page should always have dark `navigationBar` (since this screen supports only dark theme);
- when user is logged in, then we should manage `navigationBar` color depending on theme.

To achieve first item (and partially second) I changed `styles.xml` for SplashScreen theme and added default `navigationBar` color as `#000`.

To change color in runtime we need to use additional libs. Mainly I was considering two variants:
- [react-native-navigation-bar-color](https://www.npmjs.com/package/react-native-navigation-bar-color) - was a pretty popular package back in times, but at the moment doesn't receive constant updates, popularity is moving down, most likely this package doesn't support Fabric/bridgeless mode etc.
- [expo-navigation-bar](https://www.npmjs.com/package/expo-navigation-bar) - offers extended API, has 40k weekly downloads (4x more popular than a previous lib), has Fabric support, constantly receives updates.

So the choice was pretty obvious - I chose `expo-navigation-bar` (we probably need to open an issue for new library).

### Fixed Issues

$ https://github.com/Expensify/App/issues/51673
PROPOSAL: N/A

### Tests

Perform tests at least on two platforms, for example Android 9 (old and popular) and Android 14 (new and popular).

- open app and be sure navigation bar color is green
- be sure login page is opened with dark navigation bar
- login into app and be sure navigation bar has the same theme as main app
- go to settings -> preferences -> theme and toggle all themes. Be sure that nav bar color is the same as main app
- log out and be sure navigation bar color is dark

### Offline tests

N/A

### QA Steps

Perform tests at least on two platforms, for example Android 9 (old and popular) and Android 14 (new and popular).

- open app and be sure navigation bar color is green
- be sure login page is opened with dark navigation bar
- login into app and be sure navigation bar has the same theme as main app
- go to settings -> preferences -> theme and toggle all themes. Be sure that nav bar color is the same as main app
- log out and be sure navigation bar color is dark

> Highly desirable to test at least one Samsung (new and popular) device. I tested on S21 Ultra with Android 14.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/c3b01429-6f52-42bd-9c67-90451be70bb3

https://github.com/user-attachments/assets/45418634-73c0-4285-ac1a-199eade68285

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

image

</details>

<details>
<summary>MacOS: Desktop</summary>

image

</details>

### CODE REVIEW

**Comment by kirillzyusko** [comment]
Opened an issue with library review here: https://github.com/Expensify/App/issues/51957

File: android/app/src/main/res/values/styles.xml
Comment (shawnborton) [inline android/app/src/main/res/values/styles.xml:15]: Just curious, at what point do we want black and why?

**Comment by shawnborton** [comment]
Not sure if you need a review from design or not but the video generally looks good to me. Pretty wild how slow the app seems though?

**Comment by kirillzyusko** [comment]
> Pretty wild how slow the app seems though?

This is how the app runs on low end device in debug mode 🙈 In this case it's super slow 😅 But in release variant it's pretty fast for this old device!

> Not sure if you need a review from design or not but the video generally looks good to me.

Well, if you can also download the app on Android and test it, then it would definitely help! ❤️

**Comment by mananjadhav** [comment]
@kirillzyusko most of the code looks good. Can you please check one of the pending author checklist item?

File: android/app/src/main/res/values/styles.xml
Comment (kirillzyusko) [reply android/app/src/main/res/values/styles.xml:15]: This is just by default to override all automatic OS color manipulations 🙃 

But this color is used on login page - do you want to choose a different one?

telegram-cloud-photo-size-2-5280970699604879249-y

The reason why I decided that black is good enough is because below we have an illustration so monochromatic green color may look a little bit inconsistent.

Comment (shawnborton) [reply android/app/src/main/res/values/styles.xml:15]: Ah I got it. What do we use for the rest of the app when we are in dark mode?

Comment (kirillzyusko) [reply android/app/src/main/res/values/styles.xml:15]: @shawnborton `productDark100` which is `#061B09`. And in light mode we use `produceLight100` which is `#FCFBF9`

**Comment by kirillzyusko** [comment]
> most of the code looks good. Can you please check one of the pending author checklist item?

@mananjadhav done 🙌

File: android/app/src/main/res/values/styles.xml
Comment (kirillzyusko) [reply android/app/src/main/res/values/styles.xml:15]: Just to put more confidence: we use `theme.appBG`, and from the code:

```ts
const darkTheme = {
    // Figma keys
    appBG: colors.productDark100,
```

We use the same color as main background 👀

Comment (shawnborton) [reply android/app/src/main/res/values/styles.xml:15]: Cool, I would love to see what it looks like, I think staying consistent with product colors makes more sense here.

Comment (shawnborton) [reply android/app/src/main/res/values/styles.xml]: Nice, I think productDark100 looks great personally - cc @Expensify/design for a gut check here, but I say let's do that instead of black.

Comment (dannymcclain) [reply android/app/src/main/res/values/styles.xml]: I'm into it too!

Comment (dubielzyk-expensify) [reply android/app/src/main/res/values/styles.xml]: Same. Dig it

Comment (kirillzyusko) [reply android/app/src/main/res/values/styles.xml]: Cool, looks like everyone agree to use `productDark100` - I pushed these changes, so this PR is ready for review then 😊

**Comment by kirillzyusko** [comment]
@mananjadhav would you mind to review this PR?

**Comment by mananjadhav** [comment]
Just finishing a deploy blocker and then getting to this.

**Comment by kirillzyusko** [comment]
@mananjadhav any updates here?..

**Comment by mananjadhav** [comment]
I was able to test it on Android 13 but but not on Android 9. I tried my emulator keeps crashing.

**Comment by mananjadhav** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/798783e7-c809-458a-9097-08e8c95b6467

https://github.com/user-attachments/assets/1b06d339-971d-4fca-8c47-7614f01f41c7

</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by mananjadhav — APPROVED** [review]
I tested this in a few recent versions but couldn't test it on Android 9.

**Review by mountiny — CHANGES_REQUESTED** [review]
HOLDing this as we did not agree internally yet to use this library, instead, we would like to ideally just use only the piece of native code that handles the dynamic colours https://github.com/Expensify/App/issues/51957 cc @jasperhuangg

**Comment by mountiny** [comment]
@kirillzyusko Should we put this PR on hold or close it as you will be able to reuse a lot of the code once edge-to-edge support is added?

**Comment by kirillzyusko** [comment]
> Should we put this PR on hold or close it as you will be able to reuse a lot of the code once edge-to-edge support is added?

Let's close it in favour of https://github.com/Expensify/App/pull/52392

**Pull request closed without merging**
